torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -2111,10 +2111,33 @@ Tensor symeig_backward(const std::vector<torch::autograd::Variable> &grads, cons
 
   auto vh = v.conj().transpose(-2, -1);
 
+  // Based on a little experiment, the original calculation of symeig_backward
+  //   (first grad) starts to deviate when the eigenalues deviation is about
+  //   1e-10 for double precision and 1e-2 for single precision (regardless of
+  //   the matrix size).
+  // See: https://colab.research.google.com/drive/1Fyk2H8hZGTtv9OH6yfr4ys2QIK2acCL8?usp=sharing
+  // Because the threshold for single precision is too high, the new algorithm
+  //   is only applied when the datatype is double precision.
+  // Also, I found setting the eps to 1e-8 (instead of 1e-10) is required to
+  //   pass the 2nd grad check (gradgradcheck).
+  double eps = (lambda.dtype() == at::kDouble) ? 1e-8 : 0.0;
```

Comment:
@IvanYashchuk @nikitaved is there a function equivalent to torch.is_grad_enabled() that I can use here?
It is useful to have different values of eps depending on whether it requires 2nd grad or only 1st grad.

Reply:
There is indeed! `at::GradMode::is_enabled()` from the header `#include <ATen/core/grad_mode.h>`.